OnlineLibrary.API/Controllers/BookController.cs

```diff
@@ -30,102 +32,60 @@ public BookController(IBookService iBook, IAuthorService iAuthor, IMapper mapper
         [HttpGet]
         public IActionResult GetAllBooks()
         {
-            try
-            {
-                return Ok(_bookService.GetAllBooks());
-            }
-            catch (Exception e)
-            {
-                return NotFound(e.Message);
-            }
+            return Ok(_bookService.GetAllBooks());
         }
 
         // GET: api/Book/GetBooksWithFilters
         [HttpGet("{authorId}/{name}/{reservation}/{inArchieve}")]
         public IActionResult GetBooksWithFilters(int? authorId, string name, int? reservation, int? inArchieve)
         {
-            try
-            {
-                return Ok(_bookService.FilterBooks(authorId, name, reservation, inArchieve));
-            }
-            catch (Exception e)
-            {
-                return NotFound(e.Message);
-            }
+            return Ok(_bookService.FilterBooks(authorId, name, reservation, inArchieve));
         }
 
         // GET: api/Book/GetBookById/[id]
         [HttpGet("{id}")]
         public IActionResult GetBookById(int? id)
         {
-            try
-            {
-                return Ok(_bookService.GetBookById(id));
-            }
-            catch (Exception e)
-            {
-                return NotFound(e.Message);
-            }
+            return Ok(_bookService.GetBookById(id));
         }
 
         // POST:  api/Book/Create
         [HttpPost]
         public IActionResult Create([FromBody] CreateBook cBook)
         {
-            try
-            {
-                // необходимо настроить добавление с тегами!
-                List<Author> authors = _authorService.GetAuthorsByIdList(cBook.Authors);
+            // необходимо настроить добавление с тегами!
+            List<Author> authors = _authorService.GetAuthorsByIdList(cBook.Authors);
 
-                Book book = _mapper.Map<CreateBook, Book>(cBook);
-                book.Authors = authors;
-                return Ok(_bookService.CreateBook(book));
-            }
-            catch (Exception e)
-            {
-                return NotFound(e.Message);
-            }
+            Book book = _mapper.Map<CreateBook, Book>(cBook);
+            book.Authors = authors;
+            return Ok(_bookService.CreateBook(book));
         }
 
         // PUT:  api/Book/UpdateReservation/[id]
         [HttpPut("{id}")]
```

Comment:
https://www.geeksforgeeks.org/difference-between-put-and-patch-request/

Compare PUT and PATCH and decide what to use is better